core/integration/module_test.go

```diff
@@ -2286,6 +2286,192 @@ type Test struct {}
 	})
 }
 
+// TestUnbundleSDK verifies that you can implement a SDK without
+// having to implements the full interface but only the ones you want.
+// cc: https://github.com/dagger/dagger/issues/7707
+func (ModuleSuite) TestUnbundleSDK(ctx context.Context, t *testctx.T) {
+	t.Run("only codegen", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		ctr := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work/sdk").
+			With(daggerExec("init", "--name=only-codegen", "--sdk=go", "--source=.")).
+			WithNewFile("/work/sdk/main.go", `package main
+
+import (
+  "context"
+  "dagger/only-codegen/internal/dagger"
+)
+
+type OnlyCodegen struct{}
+
+func (o *OnlyCodegen) Codegen(
+  ctx context.Context,
+  modSource *dagger.ModuleSource,
+  introspectionJSON *dagger.File,
+) (*dagger.GeneratedCode, error) {
+  return dag.GeneratedCode(dag.Directory().WithNewFile("hello.txt", "Hello, world!")), nil
+}
+`).
+			WithWorkdir("/work").
+			With(daggerExec("init", "--name=test", "--sdk=./sdk", "--source=."))
+
+		t.Run("can run dagger develop", func(ctx context.Context, t *testctx.T) {
+			generatedFile, err := ctr.With(daggerExec("develop")).File("/work/hello.txt").Contents(ctx)
+
+			require.NoError(t, err)
+			require.Equal(t, "Hello, world!", generatedFile)
+		})
+
+		t.Run("explicit error on dagger call", func(ctx context.Context, t *testctx.T) {
+			_, err := ctr.With(daggerExec("call", "foo")).Sync(ctx)
+
+			requireErrOut(t, err, "the SDK does not implement the `Runtime` interface")
```

Comment:
I think the error message might be better as `"only-codegen" sdk does not support a runtime`.

Reply:
Done